qa/workunits/rbd/rbd_mirror_helpers.sh

```diff
@@ -701,7 +701,38 @@ test_status_in_pool_dir()
         grep "service: " ${status_log} && return 1
     fi
 
+    # recheck using `mirror pool status` command to stress test it.
+
+    local last_update="$(sed -nEe 's/^ *last_update: *(.*) *$/\1/p' ${status_log})"
+    test_mirror_pool_status_verbose \
+        ${cluster} ${pool} ${image} "${state_pattern}" "${last_update}" &&
     return 0
+
+    echo "'mirror pool status' test failed" >&2
+    exit 1
+}
+
+test_mirror_pool_status_verbose()
+{
+    local cluster=$1
+    local pool=$2
+    local image=$3
+    local state_pattern="$4"
+    local prev_last_update="$5"
+
+    local status_log=${TEMPDIR}/${cluster}-${pool}.mirror_status
+
+    rbd --cluster ${cluster} mirror pool status ${pool} --verbose --format xml \
+        > ${status_log}
+
+    local last_update state
+    last_update=$($XMLSTARLET sel -t -v \
+        "//images/image[name='${image}']/last_update" < ${status_log})
+    state=$($XMLSTARLET sel -t -v \
+        "//images/image[name='${image}']/state" < ${status_log})
+
+    echo "${state}" | grep "${state_pattern}" ||
+    test "${last_update}" '>' "${prev_last_update}"
```

Comment:
Why not "-ge"? Isn't this the `rbd-mirror` last update time which is throttled to every 30 or so seconds?

Reply:
"-ge" compares integers, while the last update is in "YYYY-MM-DD HH:MM:SS" format. Yes this is that last update time.

The idea is that I get the state with 'mirror image status' and then with 'mirror pool status' and compare them, expecting them to be the same. The problem is that between 'mirror image status' and 'mirror pool status'  run the image state can be updated, so I also compare last update and if they are different I consider this is such a case and I don't fail.

Reply:
... so should it be `test "${last_update}" '>' "${prev_last_update}" || test "${last_update}" '=' "${prev_last_update}"`?

Reply:
Ah, I see now what you meant: test for '>=' (which does not exists). No, I don't think we need to test for `=` for `"${last_update}" '=' "${prev_last_update}"`. Note, as the first step we check for the `state` matches `state_pattern`. And only if it does not match we look if the `last_update` is newer then we expect. Because only if it is newer it is expected to have different `state`. If I added `|| test "${last_update}" '=' "${prev_last_update}"` it would mean ok when `last_update` was not changed (status was not updated) but `state` did not match the `state` collected with `mirror image status`.

Reply:
... ok, fair enough. Thx.